### DISCUSSION TITLE
Self compiled binaries and assets

### DISCUSSION THREAD

**Discussion by DerPit** [diasurgical/DevilutionX#7333] [Q&A]
(Sorry for opening that as an issue first :o )

I have compiled the latest git version of devilutionx on my computer at home (openSUSE Tumbleweed), following the guide in the docs. No issues with that.
However, this build always insists in loading assets (lua stuff etc) from the build tree, and doesn't run if those cannot be found. The binary in the release doesn't show that behavior.
I.e., if I run (from an xterm)
<path_to_build>/devilutionx
the game runs fine.  If I copy the binary to ~/bin and run it from there I get a (graphical) warning that it cannot find (IIRC) lua/events.lua.  I can only acknowledge, and the game closes.

Running strace (for both cases) shows that it loads (or tries to) <path_to_binary>/assets/lua/....
which of course doesn't exist for the binary in ~/bin.  The binary from the official release doesn't do this, it runs without the on-disk assets directory.

I was searching whether I need some additional flags to cmake, but didn't find anything.
A hint what I'm probably doing wrong would be highly appreciated....

**Comment by AJenbo** [comment]
`assets/` is only really for development, make sure to build devilutionx.mpq and place it in a search path. You can run --verbose to see where it looks for it.

**Reply by DerPit** [reply]
Thanks AJenbo!
I was using the devilutionx.mpq from the release, it is in ~/.local/share/diasurgical/devilution/, and is found there and opened.
Does that mean it doesn't like that file, and I *always* have to build the mpq myself to match the binary?

**Reply by qndel** [reply]
devilutionx.mpq from release was up to date during release, I'm assuming you are building latest code = it can be outdated
latest devilutionx.mpq always at https://github.com/artifacts-storage/devilutionx-artifacts/blob/master/data/devilutionx.mpq

**Reply by AJenbo** [reply]
> I was using the devilutionx.mpq from the release

Then the problem is that you where using an outdated MPQ compared to the application version. If you want to install things in the way you are then you will need up to date assets for it.

**Reply by glebm** [reply]
We should probably have a `VERSION` file in devilutionx.mpq, similar to how we handle extra fonts since https://github.com/diasurgical/devilutionX/pull/6364

Sent https://github.com/diasurgical/devilutionX/pull/7337

**Reply by glebm** [reply]
@DerPit When you run `cmake --build build`, `devilutionx.mpq` is built and placed into `build/devilutionx.mpq` (assuming you have `smpq` installed, if not you can run https://github.com/diasurgical/devilutionX/blob/master/tools/build_and_install_smpq.sh)

**Reply by DerPit** [reply]
> > I was using the devilutionx.mpq from the release
> 
> Then the problem is that you where using an outdated MPQ compared to the application version. If you want to install things in the way you are then you will need up to date assets for it.

Ah, maybe this should be mentioned / pointed out in the build docs?  From what I read there, the smpq is optional if you want to build the mpq.  But obviously it usually is mandatory if you compile yourself?
So I'll see if I can install that, too :)

**Reply by AJenbo** [reply]
The assets can also be placed relative to the mpq search path (not in the assets folder )

**Reply by DerPit** [reply]
> devilutionx.mpq from release was up to date during release, I'm assuming you are building latest code = it can be outdated latest devilutionx.mpq always at https://github.com/artifacts-storage/devilutionx-artifacts/blob/master/data/devilutionx.mpq

Thanks for the pointer!
I tried that one, but it seems broken:  **Much** smaller than the previous one, and trying to use it results in an ERROR: Format error

Likely what you mentioned in the other (closed) issue thread?

**Reply by DerPit** [reply]
So I installed smpq and rebuilt everything, this time including devilutionx.mpq. The latter has a proper size, and gets accepted from the devilutionx binary if installed in a place where it can be found. :D
 Thanks for all the help! :+1:
